src/app/api/link-preview/route.ts

```diff
@@ -8,22 +7,26 @@ import urlMetaData from "url-metadata";
 export async function GET(request: NextRequest) {
   const { session } = await getUserAuth();
   if (!session) {
-    return NextResponse.json("Unauthorized", { status: 401 });
+    return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
   }
-  const url = request.nextUrl.searchParams.get("url");
+  try {
+    const url = request.nextUrl.searchParams.get("url");
 
-  const metadata = await urlMetaData(url);
+    const metadata = await urlMetaData(url);
 
-  const response = {
-    success: 1,
-    meta: {
-      title: metadata.title,
-      description: metadata.description,
-      image: {
-        url: metadata["og:image"],
+    const response = {
+      success: 1,
+      meta: {
+        title: metadata.title,
+        description: metadata.description,
+        image: {
+          url: metadata["og:image"],
+        },
       },
-    },
-  };
-
-  return NextResponse.json(response);
+    };
+    return NextResponse.json(response);
+  } catch (error) {
+    console.error(error);
+    return NextResponse.json({ error: "Server error" }, { status: 500 });
+  }
```

Comment:
1. I think using .then().catch() make this much cleaner than try catch.
2. Is `metadata["og:image"]` always a full url? I think sometimes it can be `/_static/filename.png`
3. Maybe fallback to use the favicon if they dont have og:image?

Reply:
will change when im free